packages/modeling/src/geometries/geom2/toOutlines.js

```diff
@@ -102,4 +92,28 @@ const toOutlines = (geometry) => {
   return outlines
 }
 
+// find the first counter-clockwise edge from startEdge and pop from nextEdges
+const popNextEdge = (startEdge, nextEdges) => {
+  if (nextEdges.length === 1) {
+    return nextEdges.pop()
+  }
+  const v0 = vec2.create()
```

Comment:
Isn’t this function creating a vertex for each invocation?

Reply:
Yes it is. Previously it was created only once per call to toOutlines, now it creates v0 for each step in the loop. Ideally that would not be necessary, but its hard to avoid when factoring out the `popNextSide` function.

I think it's still important to split out the `popNextSide` function even if it means an additional allocation. First, the code is more readable. Second, this is not the expensive part of toOutlines, so the impact is minimal. Third, breaking up large functions actually helps performance. I spend a lot of time in the chrome performance debugger looking at charts like the following. You can see that the v8 javascript engine will optimize certain functions. Making functions small and with narrow scope variables helps the compiler to be fast and efficient.
![trace2](https://user-images.githubusercontent.com/1766297/163538221-db3022de-38da-4997-a6e3-a09002cddd30.png)

Open to suggestions, but I felt this was worth it.

Reply:
The function is faster now so there's no worry. I just wanted to make sure that the allocation was noticed.